lib/install/validate/validator.go

```diff
@@ -501,21 +504,140 @@ func (v *Validator) certificateAuthorities(ctx context.Context, input *data.Data
 	defer trace.End(trace.Begin(""))
 
+	// Check if CAs can be loaded
+	pool := x509.NewCertPool()
+	if len(input.RegistryCAs) > 0 {
+		if !pool.AppendCertsFromPEM(input.RegistryCAs) {
+			v.NoteIssue(errors.New("Unable to load certificate authority data for registry"))
+			return
+		}
+	}
+
+	conf.RegistryCertificateAuthorities = input.RegistryCAs
+
+	// test reachability
+	insecureRegistries, whitelistRegistries, err := v.reachableRegistries(ctx, input, pool)
+	if err != nil {
+		v.NoteIssue(err)
+		return
+	}
+
 	// copy the list of insecure registries
-	conf.InsecureRegistries = input.InsecureRegistries
+	conf.InsecureRegistries = insecureRegistries
+
+	// copy the list of whitelist registries
+	conf.RegistryWhitelist = whitelistRegistries
+
+	// create vic-machine info message
+	if len(conf.InsecureRegistries) > 0 {
+		msg := "Insecure registries = "
+		for i, w := range conf.InsecureRegistries {
+			if i == 0 {
+				msg = msg + w.Host
+			} else {
+				msg = msg + ", " + w.Host
+			}
+		}
+		log.Infof(msg)
+	}
+
+	// create vic-machine info message
+	if len(conf.RegistryWhitelist) > 0 {
+		msg := "Whitelist registries = "
+		for i, w := range conf.RegistryWhitelist {
+			if i == 0 {
+				msg = msg + w.Host
+			} else {
+				msg = msg + ", " + w.Host
+			}
+		}
+		log.Infof(msg)
+	}
 
 	if len(input.RegistryCAs) == 0 {
 		return
 	}
+}
 
-	// Check if CAs can be loaded
-	pool := x509.NewCertPool()
-	if !pool.AppendCertsFromPEM(input.RegistryCAs) {
-		v.NoteIssue(errors.New("Unable to load certificate authority data for registry"))
-		return
+// Validate registries are reachable.  Secure registries that are not specified as insecure are validated with the
+// CA certs passed into vic-machine.
+func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []url.URL, whitelistRegistries []url.URL, err error) {
+	secureRegistries := input.WhitelistRegistries
+
+	// Remove intersection between insecure registries and whitelist registries from whitelist set so
+	// we can ensure we test the exclusion set with certs
+	var exist bool
+	var idx int
+	var s url.URL
+	for _, r := range input.InsecureRegistries {
+		exist = false
+		for idx, s = range secureRegistries {
+			if r.Host == s.Host {
+				exist = true
+				break
+			}
+		}
+
+		// remove the insecure registry from list of registries to get validated against certs
+		if exist {
+			secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+		}
 	}
 
-	conf.RegistryCertificateAuthorities = input.RegistryCAs
+	// Test insecure registries' reachability
+	for _, r := range input.InsecureRegistries {
+		// Make sure address is not a wildcard domain or CIDR.  If it is, do not validate.
+		if strings.HasPrefix(r.Host, "*") {
+			log.Infof("Skipping registry validation for %s", r.Host)
+			continue
+		}
+		if strings.Index(r.Path, "/") != -1 {
```

Comment:
@hickeng Yes, I added the symmetry for insecure-registry since it's goes through the same code path.